server/cla.go

```diff
@@ -54,80 +59,108 @@ func (s *Server) checkCLA(pr *model.PullRequest) {
+		mlog.Error("Unable to create the github status for for PR", mlog.Int("pr", pr.Number), mlog.Err(err))
+		return
+	}
+}
+
+func (s *Server) getCSV() ([]byte, error) {
 	resp, err := http.Get(s.Config.SignedCLAURL)
 	if err != nil {
 		mlog.Error("Unable to get CLA list", mlog.Err(err))
-		return
+		s.logToMattermost("unable to get CLA google csv file Error: ```" + err.Error() + "```")
+		return nil, err
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	resp.Body.Close()
 	if err != nil {
 		mlog.Error("Unable to read response body", mlog.Err(err))
-		return
+		s.logToMattermost("unable to read CLA google csv file Error: ```" + err.Error() + "```")
+		return nil, err
 	}
+	return body, nil
+}
 
-	claStatus := &github.RepoStatus{
-		TargetURL: github.String(s.Config.SignedCLAURL),
-		Context:   github.String("cla/mattermost"),
+func (s *Server) IsAlreadySigned(pr *model.PullRequest) bool {
+	status, err := s.GetStatus(pr, s.Config.CLAGithubStatusContext)
+	if err != nil || status == nil {
+		return false
 	}
 
-	// Get Github comments
-	comments, _, err := s.GithubClient.Issues.ListComments(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number, nil)
-	if err != nil {
-		mlog.Error("pr_error", mlog.Err(err))
-		return
-	}
+	return status.GetState() == stateSuccess
+}
 
-	lowerUsername := strings.ToLower(username)
-	tempCLA := strings.Split(string(body), "\n")
-	for _, item := range tempCLA {
-		itemCLA := strings.TrimSpace(item)
-		if strings.Compare(itemCLA, username) == 0 || strings.Compare(itemCLA, lowerUsername) == 0 || strings.Compare(strings.ToLower(itemCLA), lowerUsername) == 0 {
-			mlog.Info("will post success on CLA", mlog.String("user", username))
-			claStatus.State = github.String(stateSuccess)
-			userMsg := fmt.Sprintf("%s authorized", username)
-			claStatus.Description = github.String(userMsg)
-			_, _, errStatus := s.GithubClient.Repositories.CreateStatus(context.Background(), pr.RepoOwner, pr.RepoName, pr.Sha, claStatus)
-			if errStatus != nil {
-				mlog.Error("Unable to create the github status for for PR", mlog.Int("pr", pr.Number), mlog.Err(errStatus))
-				return
-			}
-			mlog.Info("will clean some comments regarding the CLA")
-			commentToRemove, existComment := checkCLAComment(comments, s.Config.Username)
-			if existComment {
-				mlog.Info("Removing old comment with ID", mlog.Int64("ID", commentToRemove))
-				_, err := s.GithubClient.Issues.DeleteComment(context.Background(), pr.RepoOwner, pr.RepoName, commentToRemove)
-				if err != nil {
-					mlog.Error("Unable to remove old Mattermod comment", mlog.Err(err))
-				}
-			}
-			return
+func isNameInCLAList(usersWhoSignedCLA []string, author string) bool {
+	authorLowerCase := strings.ToLower(author)
+	for _, userToTrim := range usersWhoSignedCLA {
+		user := strings.TrimSpace(userToTrim)
+		if strings.Compare(user, author) == 0 || strings.Compare(user, authorLowerCase) == 0 || strings.Compare(strings.ToLower(user), authorLowerCase) == 0 {
+			return true
```

Comment:
Nit: this can be simplified 

```suggestion
	for _, userToTrim := range usersWhoSignedCLA {
		user := strings.ToLower(strings.TrimSpace(userToTrim))
		if strings.Compare(user, author) == 0 {
			return true
```